C/services/south/include/ingest.h

```diff
@@ -30,7 +33,10 @@ class Ingest {
 public:
 	Ingest(StorageClient& storage,
 		unsigned long timeout,
-		unsigned int threshold);
+		unsigned int threshold,
+		const std::string& serviceName,
+		const std::string& pluginName,
+		ManagementClient *m_mgmtClient);
```

Comment:
Maybe we should pass it the South Service instead so that it can pick up all these items rather than having such a big set of arguments.

Reply:
Only 4th (service name) and 6th argument (ManagementClient ptr) arguments are members of SouthService class, other 4 are local variables. And those 2 also are private member variables, so their get function would need to be added and it would just reduce the number of arguments by 1.

Reply:
We can revisit this again in future and see if we can reduce the number of arguments.